langchain/src/vectorstores/typesense.ts

```diff
@@ -230,15 +239,15 @@ export class Typesense extends VectorStore {
       {}
     );
     const results = typesenseResponse.results[0].hits;
-    const hits = results?.map((hit) => hit.document) as
-      | Record<string, unknown>[]
-      | undefined;
 
-    const documents = this._typesenseRecordsToDocuments(hits).map(
-      (doc) => [doc, 0] as [Document<Record<string, unknown>>, number]
-    );
+    const hits = results?.map((hit: VectorSearchResponseHit<object>) => ({
+      document: hit?.document || {},
+      vector_distance: hit?.vector_distance || 2,
```

Comment:
why default to 2?

Reply:
According to their [Vector Search Documentation](https://typesense.org/docs/0.24.1/api/vector-search.html#nearest-neighbor-vector-search):
>  Typesense uses the cosine similarity, so this distance will be a value between 0 and 2.
>
>  - If the document's vector perfectly matches the query vector, the distance will be 0
> - If the document's vector is extremely different from the query vector, then the distance will be 2.

So I assumed that if we don't get back a vector distance, we just assume that the match is the most different.

Which makes me think about the implicit conventions in vectorstores for the `score` value. Is the expectation that a higher score normally represents a higher match? AFAIK that is the default in pinecone for example. In the case of typesense it's the opposite